packages/modules/itwin-viewer-react/src/services/iModel/IModelService.ts

```diff
@@ -4,7 +4,36 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { Id64String, OpenMode } from "@bentley/bentleyjs-core";
-import { RemoteBriefcaseConnection } from "@bentley/imodeljs-frontend";
+import { IModelHubClient, VersionQuery } from "@bentley/imodelhub-client";
+import { IModelVersion } from "@bentley/imodeljs-common";
+import {
+  IModelApp,
+  RemoteBriefcaseConnection,
+} from "@bentley/imodeljs-frontend";
+import { AuthorizedClientRequestContext } from "@bentley/itwin-client";
+
+/** determine the proper version of the iModel to open
+ * 1. If named versions exist, get the named version that contains the latest changeset
+ * 2. If no named version exists, return the latest changeset
+ */
+const getVersion = async (iModelId: string): Promise<IModelVersion> => {
+  const token = await IModelApp.authorizationClient?.getAccessToken();
+  if (token) {
+    const requestContext = new AuthorizedClientRequestContext(token);
+    const hubClient = new IModelHubClient();
+    const namedVersions = await hubClient.versions.get(
+      requestContext,
+      iModelId,
+      new VersionQuery().top(1)
```

Comment:
Will this always return something?  What if it's a new iModel that hasn't had any changesets applied yet?

Reply:
I see you're catching it below, but maybe we could provide a better error message.

Reply:
I think this is ok. This is essentially a WSG query, so it should always return an instance array.

`new VersionQuery().top(1)` is just a fancy way to add $top=1 to the query. 

We are checking length below this and just returning `IModelVersion.latest()` if namedversions.length === 0, which is the default anyways so it should handle new iModels.